### ISSUE TITLE
record column not supported

### ISSUE THREAD

**Issue by sinkog** [godmodelabs/flora-sql-parser#23]
record column not supported

`x=parser.parse('SELECT cv,i FROM voucher cv JOIN item i ON i.dbid=cv.dbid AND i.cvoid=cv.void AND i.deleted=0')`

**Comment by mbaumgartl** [comment]
It doesn't seem to be a problem with the parser. I can parse the query into an AST and convert it back to SQL without any problems.

Can you explain your issue more detailed?

**Comment by sinkog** [comment]
Naturally!
There is a complicated aggregate query with which to retrieve data. Let's say 30 fields would be the return value in the basic case. However, not all cases should have all the fields in the query just as a basic template would live. Say I need 10 fields and of course the related tables. If the parser refers to the record as a field name, which is also a table alias, then the pendant table can be manipulated by tapping.
Currently, the parsed data is:
`Object
￼columns: Array (1 item)
￼0: Object
as: null
￼expr: Object {column: "cv", table: null, type: "column_ref"}
...`

For me it would be right:
`Object ￼columns: Array (1 item) ￼0: Object as: null ￼expr: Object {column: "cv", table: "cv", type: "column_ref"}...`

I know the alias of the table can be a field that we want to take. (now i can trick but i do not like the solution)

**Comment by mbaumgartl** [comment]
Does the (original) query work on a real database?

If you want to select all columns from a table, the syntax would be
`SELECT t.* FROM t`
Otherwise the identfiers are treated as column names.

**Comment by sinkog** [comment]
At present I have highlighted a problem I could not solve. The original query:
`WITH ss AS (
SELECT cv,i,il,ct,tcs
FROM voucher cv
JOIN item i ON i.dbid=cv.dbid AND i.cvoid=cv.void AND i.deleted=0
JOIN item_lot il ON il.dbid=i.dbid AND il.itemid=i.itemid AND il.deleted=0
JOIN cikktorzs ct ON ct.ocsid=i.ocsid
JOIN termekcsoport tcs ON tcs.tcskod=ct.tcskod
WHERE cv.votp='VSZM' AND cv.perfdate>'2017-09-01' AND cv.perfdate<'2017-09-10' AND il.objid=136157 AND tcs.tcskod='H'
)
SELECT jsonb_pretty(json_agg(json_build_object('void', v.void, 'perfdate', v.perfdate, 'items', item.j))::jsonb) as j
FROM (SELECT (x.cv).* FROM (SELECT (ss.cv) as cv FROM ss GROUP BY cv) x) v
JOIN LATERAL (
	SELECT json_agg(json_build_object('itemid', i.itemid, 'qty', i.qty,'cikk',cikktorzs.j, 'lots', item_lot.j)) as j FROM 
	(SELECT (x.i).* FROM (SELECT (ss.i) as i FROM ss GROUP BY i) x) i
	JOIN LATERAL (
		SELECT json_agg(json_build_object('objid', il.objid, 'qty', il.qty)) as j 
		FROM (SELECT (x.il).* FROM (SELECT (ss.il) as il FROM ss GROUP BY il) x) il 
		WHERE il.itemid=i.itemid AND il.dbid=i.dbid 
	    ) item_lot ON true
	JOIN LATERAL (
		SELECT json_build_object('cikkszam',ct.cikkszam,'megnev',ct.megnev,'termekcsoport',termekcsoport.j) as j 
		FROM (SELECT (x.ct).* FROM (SELECT (ss.ct) as ct FROM ss GROUP BY ct) x) ct
		JOIN LATERAL (
			SELECT json_build_object('tcskod',tcskod,'megnev',megnev) as j 
			FROM (SELECT (x.tcs).* FROM (SELECT (ss.tcs) as tcs FROM ss GROUP BY tcs) x) tcs WHERE tcs.tcskod=ct.tcskod
		    ) termekcsoport ON true  
		WHERE ct.ocsid=i.ocsid
	    ) cikktorzs ON true
	WHERE i.cvoid=v.void AND i.deleted=0 AND item_lot.j is not null
    ) item ON true
WHERE item.j is not null`
In this case, the named record fields are needed

**Comment by mbaumgartl** [comment]
:astonished:

**Comment by mbaumgartl** [comment]
Looks like a Postgres query. I'm pretty sure the parser can't handle it at the moment.

**Comment by sinkog** [comment]
Yes this is a postgresql query. What the current parser treats is almost all the 'LITERAL JOIN' you do not know but this is the least. What is the original problem is the settlement of the right name and table.
The solution to this is that the alias names for the tables should be searched in the built-in object and as in the SELECT section, the name matches the given table's table alias, as well as a table reference.
At this time I did not find a node module for my goals. Maybe that's the closest thing to my goals. I would like to query a more complex database by GraphQL standard and properly document its interface. To do this, a PostgreSQL query consolidation / optimization is indispensable, and I am using a new or new project or using a project of someone else.
I think, but I have not tested it far enough, your query parsers would almost fit my ideas.
In its present state after a post processing I looked at the fault / deficiency raised can be remedied. Question is why why not have your other project even better?
